src/main/scala/mesosphere/marathon/Main.scala

```diff
@@ -27,7 +30,32 @@ class MarathonApp extends App {
     val client = new ZooKeeperClient(
       Amount.of(conf.zooKeeperSessionTimeout().toInt, Time.MILLISECONDS),
       conf.zooKeeperHostAddresses.asJavaCollection
-    )
+    ) {
+      override def shouldRetry(e: KeeperException): Boolean = {
+        log.error(s"Got ZooKeeper exception: $e")
```

Comment:
I would put the exception into an extra argument such that you can see stack trace.